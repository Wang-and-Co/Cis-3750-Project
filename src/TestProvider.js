/* eslint-disable react/destructuring-assignment */
import * as React from 'react';
import { createTheme } from '@mui/material';
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles';
// import defaultTheme from '../../src/renderer/data/theme';

export default class TestProvider extends React.Component {
  disableTransitions = {
    defaultProps: {
      disablePortal: true,
      hideBackdrop: true,
      TransitionComponent: ({ children }) => children,
    },
  };

  render() {
    const theme = createTheme({
      //   ...defaultTheme,
      // disable Transitions when runing tests, throws errors otherwise.
      components: {
        MuiDialog: this.disableTransitions,
        MuiModal: this.disableTransitions,
      },
    });
    return (
      // eslint-disable-next-line react/jsx-no-comment-textnodes
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={theme}>{this.props.children}</ThemeProvider>
      </StyledEngineProvider>
    );
  }
}
