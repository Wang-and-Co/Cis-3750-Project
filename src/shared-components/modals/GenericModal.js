import { Typography, Stack, Button, Dialog, Box } from '@mui/material';
import { PropTypes } from '@mui/material';
const style = {
  // position: 'absolute',
  top: '50%',
  left: '50%',
  // transform: 'translate(-50%, -50%)',
  width: 400,
  // bgcolor: 'background.paper',
  border: '2px solid #000',
  boxShadow: 24,
  p: 4,
};

/**
 * @typedef GenericModalProps
 * @type {object}
 * @property {string} name
 * @property {string} title
 * @property {string} body
 * @property {string} id
 * @property {function} onClose
 * @property {boolean} open
 */

/**
 * A generic modal that should be a base for other modals
 * @param {GenericModalProps} props
 * @returns
 */
const GenericModal = ({ name, title, body, id, onClose, open }) => (
  <Dialog name={name} open={open}>
    <Box sx={style}>
      {title && (
        <Typography id={`modal-${id}-title`} variant="h3">
          {title}
        </Typography>
      )}
      {body}
      <Stack spacing={2} direction="row-reverse">
        <Button variant="contained" onClick={onClose}>
          Close
        </Button>
      </Stack>
    </Box>
  </Dialog>
);

export default GenericModal;
