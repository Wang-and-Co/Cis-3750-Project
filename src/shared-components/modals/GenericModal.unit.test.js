import { render } from '@testing-library/react';
import GenericModal from './GenericModal';
import { create } from 'react-test-renderer';
import TestProvider from '../../TestProvider';

jest.mock('react-dom', () => {
  const original = jest.requireActual('react-dom');

  return {
    ...original,
    createPortal: (node) => node,
  };
});

const props = {
  name: 'test',
  title: 'title',
  body: 'desc',
  id: 'id',
  onClose: jest.fn(),
  open: true,
};

let genericModalRenderer;
let genericModalInstance;

beforeEach(async () => {
  await renderGenericModal(props);
});

afterEach(() => {
  genericModalRenderer.unmount();
});
describe('Generic Modal', () => {
  it('renders correctly', () => {
    expect(genericModalRenderer.toJSON()).toMatchSnapshot();
  });
  it('passes the correct props', () => {
    expect(getModalProp('name')).toBe(props.name);
    expect(getModalProp('title')).toBe(props.title);
    expect(getModalProp('body')).toBe(props.body);
    expect(getModalProp('id')).toBe(props.id);
    expect(getModalProp('open')).toBe(props.open);
  });
});

const getModalProp = (prop) =>
  genericModalInstance.findByType(GenericModal).props[prop];

const renderGenericModal = async (editorProps) => {
  genericModalRenderer = await create(
    <TestProvider>
      <GenericModal {...editorProps}></GenericModal>
    </TestProvider>,
  );
  genericModalInstance = genericModalRenderer.root;
};
