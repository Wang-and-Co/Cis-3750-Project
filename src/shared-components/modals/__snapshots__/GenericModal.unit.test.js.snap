// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generic Modal renders correctly 1`] = `
<div
  TransitionComponent={[Function]}
  className="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  name="test"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
>
  <div
    data-testid="sentinelStart"
    onFocus={[Function]}
    tabIndex={0}
  />
  <div
    className="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    onMouseDown={[Function]}
  >
    <div
      aria-labelledby=":r0:"
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <div
        className="MuiBox-root css-1qqrgcw"
      >
        <h3
          className="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
          id="modal-id-title"
        >
          title
        </h3>
        <div
          className="MuiBox-root css-1w2p18d"
        >
          desc
        </div>
      </div>
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    onFocus={[Function]}
    tabIndex={0}
  />
</div>
`;
